crates/turborepo-lib/src/engine/mod.rs

```diff
@@ -110,4 +114,110 @@ impl Engine<Built> {
                 .collect(),
         )
     }
+
+    pub fn validate(
+        &self,
+        package_graph: &PackageGraph,
+        concurrency: u32,
+    ) -> Result<(), Vec<ValidateError>> {
+        // TODO(olszewski) once this is hooked up to a real run, we should
+        // see if using rayon to parallelize would provide a speedup
+        let (persistent_count, mut validation_errors) = self
+            .task_graph
+            .node_indices()
+            .map(|node_index| {
+                let TaskNode::Task(task_id) = self
+                    .task_graph
+                    .node_weight(node_index)
+                    .expect("graph should contain weight for node index")
+                else {
```

Comment:
Didn't even know you could apply this syntax on enums like that. Neat.

Reply:
I too learned that this was possible from @NicholasLYang and I've been in love ever since.